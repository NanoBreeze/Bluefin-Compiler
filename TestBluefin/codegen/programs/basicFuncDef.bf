/*
Verifies basic function definitions, especially return type and param types
TODO: handle string and user-defined struct type
*/

bool hi() { }

void hey(bool a, float b, int c) {
}

float foo(float a) {}

int bar() {}