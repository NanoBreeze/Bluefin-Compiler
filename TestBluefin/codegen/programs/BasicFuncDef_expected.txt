; ModuleID = 'BasicFuncDef.bf'
source_filename = "BasicFuncDef.bf"

define i1 @hi() {
entry:
}

define void @hey(i1 %a, float %b, i32 %c) {
entry:
}

define float @foo(float %a) {
entry:
}

define i32 @bar() {
entry:
}
